### PR TITLE
Feature/calendar

### PR DESCRIPTION
**Pull request by maxahn**
Refactoring, including fixing some of the many many lint errors which mostly had to do with accessibility issues. they aren't all fixed but it's a start. we really need to fix a few of the lint settings though, it doesn't seem to be caught up completely to all the new ES6 features e.g. 
where you don't have to repeat the property name when assigning it to an object with the same attribute name
const { someProperty } = this.props;
const newObj = { someProperty };

### CODE REVIEW

File: src/components/TaskList/components/TaskItem/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+  countMangoDonations = () => {
+    const { mangoTransactions } = this.props.task;
+    return mangoTransactions.reduce((acc, curr) => {
+      return acc + curr.mangoAmount;
+    }, 0);
+  };
+
+  updateModal = () => {
+    const v = this.props.updateTasks;
```
Comment (rhiknow55) [inline src/components/TaskList/components/TaskItem/index.js:70]: seems extra

File: src/components/TaskList/components/TaskItem/index.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  };
+
+  updateDueDate = (dueDate) => {
+    const { updateTask } = this.props;
+    const updatedTask = { ...this.props.task, dueDate };
+    updateTask(updatedTask);
+  };
+
+  deleteTask = () => {
+    const { deleteTask, task } = this.props;
```
Comment (rhiknow55) [inline src/components/TaskList/components/TaskItem/index.js:94]: task unused

Comment (AamirL1011) [reply src/components/TaskList/components/TaskItem/index.js:94]: Isn't "task" being passed into the deleteTask function?

File: src/components/TaskList/components/TaskItem/index.js (hunk 1 above)
Comment (maxahn) [reply src/components/TaskList/components/TaskItem/index.js:70]: it's mohammad's and I wasn't sure if he was going to use it so I left it alone for now

File: src/components/TaskList/components/TaskItem/index.js (hunk 2 above)
Comment (maxahn) [reply src/components/TaskList/components/TaskItem/index.js:94]: task is used to get id though?

Comment (rhiknow55) [reply src/components/TaskList/components/TaskItem/index.js:94]: oh i'm blind, mb

**Pull request merged**